pxc/jenkins/pxc-package-testing-test-yash.groovy

```diff
@@ -0,0 +1,310 @@
+                KEYPATH_COMMON="/home/ec2-user/.cache/molecule/pxc80-common/centos-6/ssh_key-us-west-2"
+            else
+                echo "OS Not yet in list of Keypath setup"
+            fi
+
+            echo \"printing path of bootstrap \$KEYPATH_BOOTSTRAP\"
+            echo \"printing path of common  \$KEYPATH_COMMON\"
+            echo \"printing user \$SSH_USER\"
+
+            Bootstrap_Instance=\$(cat \${BOOTSTRAP_INSTANCE_PUBLIC_IP} | jq -r .[0] | jq [.instance] | jq -r .[])
+            Bootstrap_Instance_Public_IP=\$(cat \${BOOTSTRAP_INSTANCE_PUBLIC_IP} | jq -r .[0] | jq [.public_ip] | jq -r .[])
+            
+            export ip_env=\$Bootstrap_Instance
+            echo "\n \$Bootstrap_Instance ansible_host=\$Bootstrap_Instance_Public_IP  ansible_ssh_user=\$SSH_USER ansible_ssh_private_key_file=\$KEYPATH_BOOTSTRAP ansible_ssh_common_args='-o StrictHostKeyChecking=no' ip_env=\$Bootstrap_Instance" > ${WORKSPACE}/package-testing/molecule/pxc/${version}-bootstrap/playbooks/inventory
+
+            export ip_env=\$Common_Instance_PXC2
+            Common_Instance_PXC2=\$(cat \${COMMON_INSTANCE_PUBLIC_IP} | jq -r .[0] | jq [.instance] | jq -r .[])
+            Common_Instance_PXC2_Public_IP=\$(cat \${COMMON_INSTANCE_PUBLIC_IP} | jq -r .[0] | jq [.public_ip] | jq -r .[])
+
+            echo "\n \$Common_Instance_PXC2 ansible_host=\$Common_Instance_PXC2_Public_IP   ansible_ssh_user=\$SSH_USER ansible_ssh_private_key_file=\$KEYPATH_COMMON ansible_ssh_common_args='-o StrictHostKeyChecking=no'  ip_env=\$Common_Instance_PXC2" > ${WORKSPACE}/package-testing/molecule/pxc/${version}-common/playbooks/inventory
+
+            export ip_env=\$Common_Instance_PXC3
+            Common_Instance_PXC3=\$(cat \${COMMON_INSTANCE_PUBLIC_IP} | jq -r .[1] | jq [.instance] | jq -r .[])
+            Common_Instance_PXC3_Public_IP=\$(cat \${COMMON_INSTANCE_PUBLIC_IP} | jq -r .[1] | jq [.public_ip] | jq -r .[])
+
+            echo "\n \$Common_Instance_PXC3 ansible_host=\$Common_Instance_PXC3_Public_IP   ansible_ssh_user=\$SSH_USER ansible_ssh_private_key_file=\$KEYPATH_COMMON ansible_ssh_common_args='-o StrictHostKeyChecking=no'  ip_env=\$Common_Instance_PXC3" >> ${WORKSPACE}/package-testing/molecule/pxc/${version}-common/playbooks/inventory
+            """
+
+}
+
+void runlogsbackup(String version) {
+    def awsCredentials = [
+        sshUserPrivateKey(
+            credentialsId: 'MOLECULE_AWS_PRIVATE_KEY',
+            keyFileVariable: 'MOLECULE_AWS_PRIVATE_KEY',
+            passphraseVariable: '',
+            usernameVariable: ''
+        ),
+        aws(
+            accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+            credentialsId: '5d78d9c7-2188-4b16-8e31-4d5782c6ceaa',
+            secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+        )
+    ]
+
+    withCredentials(awsCredentials) {
+        sh """
+            source venv/bin/activate
+            export MOLECULE_DEBUG=0
+            export install_repo="${params.INSTALL_REPO}"
```

Comment:
Why do we need to pass instaall_repo for runlogsbackup? AFAICS it is not used here. And it seems separate python venv here is also not required for this task.

Reply:
Hi,

Will need to keep the source venv/bin/activate as the archiving fails if it is not present